saleor/graphql/context.py

```diff
@@ -45,18 +44,18 @@ def set_app_on_context(request: SaleorContext):
         request.app = load_app(request)
 
 
-def get_user(request: SaleorContext) -> Optional[UserType]:
+def get_user(request: SaleorContext) -> UserType:
     if not hasattr(request, "_cached_user"):
         request._cached_user = cast(UserType, authenticate(request=request))
     return request._cached_user
 
 
 def set_auth_on_context(request: SaleorContext):
     if hasattr(request, "app") and request.app:
-        request.user = AnonymousUser()
+        request.user = SimpleLazyObject(lambda: None)  # type: ignore
```

Comment:
I see what you did there :smirk:

Reply:
Why can't we assign `None` here directly?

Reply:
As, once I get `SimpleLazyObject` and once I get the `None` instance, it was hard to handle, I cannot use the methods for lazy objects without checking the type. IMO, we should always return the object of the same type in the context.

Reply:
If this works it's fine for now. I'm changing this to dataloader anyway :wink: